libs/ledger/src/transaction_verifier.cpp

```diff
@@ -31,6 +35,77 @@ static const std::chrono::milliseconds POP_TIMEOUT{300};
 
 namespace fetch {
 namespace ledger {
+namespace {
+using telemetry::Registry;
+
+std::string CreateMetricName(std::string const &prefix, std::string const &name)
+{
+  // build up the basic name
+  std::string metric_name = "ledger_" + prefix + '_' + name;
+
+  // ensure there are no invalid characterss
+  std::transform(metric_name.begin(), metric_name.end(), metric_name.begin(), [](char c) -> char {
+    if (c == '-')
+    {
+      return '_';
+    }
+    else
+    {
+      return static_cast<char>(std::tolower(c));
```

Comment:
Include a check using `std::isalnum` to discard non-alphanumerics?

Reply:
Discarding of the characters would require a little more effort here. Ultimately this is more of a connivence thing than a necessity so I am inclined to leave it as is at the moment.

Reply:
sure